src/core/OIDNDenoiser.js

```diff
@@ -0,0 +1,298 @@
+
+		this.renderer.resetState();
+		if ( ! this.outputTexture ) this.outputTexture = this.createOutputTexture();
+		// inject the webGLTexture into the texture
+		this.denoisedTexture = this.injectWebGLTexture( this.outputTexture, denoisedWebGLTexture );
+		// mark as complete and setup the renderer
+		this.isDenoising = false;
+		// we use this for the fade in
+		this.denoiserFinished = performance.now();
+		this.handleProgress( 1 );
+		return this.denoisedTexture;
+
+	}
+
+	// render the blended output
+	renderOutput( bypassTextureName ) {
+
+		const bypassTexture = this.auxTextures[ bypassTextureName ];
+
+		if ( ! this.denoisedTexture ) return;
+
+		this.quad.material = this.blendToCanvasMaterial;
+		this.blendToCanvasMaterial.map = this.denoisedTexture;
+		this.blendToCanvasMaterial.opacity = Math.min( ( performance.now() - this.denoiserFinished ) / this.fadeTime, 1 );
+
+		// Lets us see the aux textures
+		if ( bypassTexture ) {
+
+			this.blendToCanvasMaterial.map = bypassTexture;
+			this.blendToCanvasMaterial.opacity = 1;
+
+		}
+
+		const currentAutoClear = this.renderer.autoClear;
+		this.renderer.autoClear = false;
+		this.quad.render( this.renderer );
+		this.renderer.autoClear = currentAutoClear;
+
+	}
+
+	// because of size issues we need to create one when we change size
+	createConversionRenderTarget( width, height ) {
+
+		// if one exists destroy it
+		if ( this.conversionRenderTarget ) this.conversionRenderTarget.dispose();
+
+		// todo Probably a better setting with dpr
+		this.conversionRenderTarget = new WebGLRenderTarget( width, height );
+		this.conversionRenderTarget.colorspace = SRGBColorSpace;
+		this.conversionRenderTarget.texture.colorspace = SRGBColorSpace;
+
+	}
+
+	// The plain texture is raw without toneMapping this is more like what renders to canvas
+	getLinearToSRGBTexture( pathtracedTexture, target ) {
+
+		const oldRenderTarget = this.renderer.getRenderTarget();
+		this.quad.material = this.linearToSRGBMaterial;
+		this.linearToSRGBMaterial.map = pathtracedTexture;
+		this.renderer.setRenderTarget( target );
+		this.quad.render( this.renderer );
+		this.renderer.setRenderTarget( oldRenderTarget );
+		return target.texture;
+
+	}
+
+	// create an output texture we can inject to
+	createOutputTexture() {
+
+		const tempRT = new WebGLRenderTarget( this.denoiser.width, this.denoiser.height );
+		// render the quad to the texture
+		const oldRenderTarget = this.renderer.getRenderTarget();
+		this.renderer.setRenderTarget( tempRT );
+		this.quad.render( this.renderer );
+		this.renderer.setRenderTarget( oldRenderTarget );
+
+		// get the texture out of the tempRT
+		const texture = tempRT.texture;
+		// dispose?
+		return texture;
+
+	}
+
+	abort() {
+
+		this.denoiser.abort();
+		this.handleProgress( 0 );
+
+	}
+
+	//* Listeners
+
+	handleProgress( progress ) {
+
+		let outProgress = progress * 100;
+		// we only return 100 when isDenoised is true but the denoiser progress is for tiling
```

Comment:
What does this mean exactly? The denoiser package is returning 100% completion but it's not actually done?

Reply:
Right now the denoiser package reports on the percentage of the tiler and not as a whole. The tiler is the main process that does not block the thread but I will look into adding other progress callbacks if possible.